### ISSUE TITLE
Can't compile with GHC-9.0.1

### ISSUE THREAD

**Issue by LeventErkok** [haskell-CI/haskell-ci#494]
```
$  ghc --version
The Glorious Glasgow Haskell Compilation System, version 9.0.1
$ cabal new-install haskell-ci
Resolving dependencies...
cabal: Could not resolve dependencies:
[__0] trying: haskell-ci-0.10.3 (user goal)
[__1] next goal: base (dependency of haskell-ci)
[__1] rejecting: base-4.15.0.0/installed-4.15.0.0 (conflict: haskell-ci =>
base>=4.10 && <4.15)
[__1] rejecting: base-4.14.1.0, base-4.14.0.0, base-4.13.0.0, base-4.12.0.0,
base-4.11.1.0, base-4.11.0.0, base-4.10.1.0, base-4.10.0.0, base-4.9.1.0,
base-4.9.0.0, base-4.8.2.0, base-4.8.1.0, base-4.8.0.0, base-4.7.0.2,
base-4.7.0.1, base-4.7.0.0, base-4.6.0.1, base-4.6.0.0, base-4.5.1.0,
base-4.5.0.0, base-4.4.1.0, base-4.4.0.0, base-4.3.1.0, base-4.3.0.0,
base-4.2.0.2, base-4.2.0.1, base-4.2.0.0, base-4.1.0.0, base-4.0.0.0,
base-3.0.3.2, base-3.0.3.1 (constraint from non-upgradeable package requires
installed instance)
[__1] fail (backjumping, conflict set: base, haskell-ci)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, haskell-ci
```

**Comment by sjakobi** [comment]
The latest 0.12 release has compatible bounds, but some of the dependencies aren't compatible yet, e.g. https://github.com/haskell-hvr/cryptohash-sha256/issues/12.

**Comment by jwaldmann** [comment]
(probably the same thing?) I am seeing this error
```
Run if [ $((HCNUMVER >= 90000)) -ne 0 ] ; then HLINTVER=$(cd /tmp && (${CABAL} v2-install -v $ARG_COMPILER --dry-run hlint  --constraint='hlint >=3.2 && <3.3' |  perl -ne 'if (/\bhlint-(\d+(\.\d+)*)\b/) { print "$1"; last; }')); echo "HLint version $HLINTVER" ; fi
Warning: Package db /github/home/.cabal/store/ghc-9.0.1/package.db does not
exist yet

cabal: Could not resolve dependencies:
[__0] next goal: hlint (user goal)
[__0] rejecting: hlint-3.3 (constraint from command line flag requires >=3.2
&& <3.3)
...
```
(for https://github.com/jwaldmann/ersatz/tree/polar-runBit - I did `haskell-ci regenerate` for an existing `.github/workflows/haskell-ci.yml`, after adding `tested-with: ... GHC==9.0.1` in the cabal file)

**Comment by phadej** [comment]
@jwaldmann note the `--constraint='hlint >=3.2 && <3.3' ` in generate code.

default HLint version in released `haskell-ci-0.12` is HLint-3.2.

See `hlint-version` configuration option.
